WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift

```diff
@@ -609,7 +667,31 @@ extension OrderDetailsDataSource {
 
             let row: Row = .refundedProducts
 
-            return Section(title: Title.refundedProducts, row: row)
+            return Section(category: .refundedProducts, title: Title.refundedProducts, row: row)
+        }()
+
+        let shippingLabels: [Section] = {
+            let shippingLabels = resultsControllers.shippingLabels
+            guard shippingLabels.isNotEmpty else {
+                return []
+            }
+
+            self.shippingLabels = shippingLabels
```

Comment:
This assignment looks like an unwanted side-effect.
Is it not possible to use `resultsControllers.shippingLabels` directly?

Reply:
the reason I set this is so that the `shippingLabels` value is fixed until the next time `reloadSections` is called. because the number of shipping label cards and its product rows depend on the value, I personally felt it's safer to have a "fixed" value rather than a variable from FRC (even though it should trigger `reloadSections` on any change, but FRC is also not fully within our control). lemme know what you think!

Reply:
I think that it's good but what I find strange is for the mutation to happen on `let shippingLabels: [Section] = {` creation. I wouldn't expect that variable initialization to mutate some bigger state of `self`.

Maybe something like:
```swift
self.shippingLabels = shippingLabels
let shippingLabels: [Section] = {
    // code
}
```
could express the intention better!

Reply:
yea great point! I update this in https://github.com/woocommerce/woocommerce-ios/pull/3188/commits/f41907f2b667a0dcb1f476908cfa772d4a68a170